README.md

```diff
@@ -202,17 +202,30 @@ apply plugin: 'app.cash.molecule'
-The clock is used to determine when recomposition occurs and a new value is produced.
+Whenever Jetpack Compose recomposes, it always waits for the next frame before beginning its work.
+It is dependent on a `MonotonicFrameClock` in its `CoroutineContext` to know when a new frame is sent.
 
-On Android, [`AndroidUiDispatcher.Main`](https://cashapp.github.io/molecule/docs/latest/molecule-runtime/app.cash.molecule/-android-ui-dispatcher/-companion/-main.html) can be used for running your composables on the main thread with recomposition synchronized to the frame rate.
-For any other rate or to recompose on a background thread, create a [`BroadcastFrameClock`](https://developer.android.com/reference/kotlin/androidx/compose/runtime/BroadcastFrameClock) and a timer to invoke its `sendFrame` function at your desired rate.
+So does Molecule: values won't be produced until a frame is sent and recomposition occurs. But Molecule has some other tricks up its sleeve, too.
+
+* `RecompositionClock.ContextClock` behaves like Jetpack Compose: it will fish the `MonotonicFrameClock` out of the calling `coroutineContext` and use it for recomposition.
+  If there is no `MonotonicFrameClock`, it will throw an exception.
+* `RecompositionClock.Immediate` will construct an immediate clock.
+This clock will produce a frame whenever the enclosing flow is ready to receive an item.
+(This is always the case for a `StateFlow`.)
+
+`ContextClock` is useful with Android's [`AndroidUiDispatcher.Main`](https://cashapp.github.io/molecule/docs/latest/molecule-runtime/app.cash.molecule/-android-ui-dispatcher/-companion/-main.html).
+`Main` has a built-in `MonotonicFrameClock` that is synchronized with the frame rate of the device.
+A Molecule run on `Main` with `ContextClock` will run in lock step with the frame rate, too. Nifty! `ContextClock` may also be used to implement your own frame rate with `BroadcastFrameClock`.
+
+`Immediate`, on the other hand, can be used where no clock is available at all without any additional wiring.
+It may be used for unit testing, or for running molecules off the main thread.
 
 ### Testing
 
-While the created `StateFlow`s and `Flow`s can be tested normally, the use of the frame clock to control recomposition makes it harder than it should be.
-The 'molecule-testing' dependency provides a `testMolecule` function which simplifies your test code by managing the threading, coroutine scope, and frame clock for you.
+There are two options for testing:
+
+1. Use `RecompositionClock.Immediate` and test using [Turbine](https://github.com/cashapp/turbine/).
+2. Use the `molecule-testing` dependency. `molecule-testing` provides a `testMolecule` function which simplifies your test code by managing the threading, coroutine scope, and frame clock for you.
```

Comment:
Is there any reason to choose one or the other?

Reply:
Matt's argument was that you might actually want the batching behavior of a manual clock to more accurately simulate what happens in production code when using something like the main clock. I don't love having two choices but would definitely like to steer people. 

Immediate == you get more updates than you would in production but it's deterministic(er) for testing

Context == you control batching which is useful for...... simulating closer to production with multiple state changes per recomposition?